ui/src/redux/modules/aggregationWs/fetchAggregation.js

```diff
@@ -0,0 +1,272 @@
+    const out = state
+      .setIn([new Map({
+        pipeline,
+        timeIntervalSinceToday,
+        timeIntervalUnits,
+        ...(timeIntervalSincePreviousTimeInterval ? { timeIntervalSincePreviousTimeInterval } : {})
+      }), 'requestState'], IN_PROGRESS);
+    return out;
+  },
+
+  reduceSuccess: (state, {
+    pipeline,
+    timeIntervalSinceToday,
+    timeIntervalUnits,
+    timeIntervalSincePreviousTimeInterval,
+    result
+  }) => {
+    if (!pipeline) {
+      return state;
+    }
+
+    const x = state
+      .setIn([new Map({
+        pipeline,
+        timeIntervalSinceToday,
+        timeIntervalUnits,
+        ...(timeIntervalSincePreviousTimeInterval ? { timeIntervalSincePreviousTimeInterval } : {})
+      }), 'requestState'], COMPLETED);
+
+    if (result.get('result') === null) {
+      return x;
+    }
+
+    const out = x.setIn([new Map({
+      pipeline,
+      timeIntervalSinceToday,
+      timeIntervalUnits,
+      ...(timeIntervalSincePreviousTimeInterval ? { timeIntervalSincePreviousTimeInterval } : {})
+    }), 'result'], result.getIn(['result']));
+
+    return out;
+  },
+
+  reduceFailure: (state, {
+    pipeline,
+    timeIntervalSinceToday,
+    timeIntervalUnits,
+    timeIntervalSincePreviousTimeInterval,
+    err
+  }) => state
+    .setIn([new Map({
+      pipeline,
+      timeIntervalSinceToday,
+      timeIntervalUnits,
+      ...(timeIntervalSincePreviousTimeInterval ? { timeIntervalSincePreviousTimeInterval } : {})
+    }), 'requestState'], FAILED)
+    .setIn([new Map({
+      pipeline,
+      timeIntervalSinceToday,
+      timeIntervalUnits,
+      ...(timeIntervalSincePreviousTimeInterval ? { timeIntervalSincePreviousTimeInterval } : {})
+    }), 'error'], err),
+
+  reduceComplete: (state, {
+    pipeline,
+    timeIntervalSinceToday,
+    timeIntervalUnits,
+    timeIntervalSincePreviousTimeInterval
+  }) => {
+    const out = state
+    .setIn([new Map({
+      pipeline,
+      timeIntervalSinceToday,
+      timeIntervalUnits,
+      ...(timeIntervalSincePreviousTimeInterval ? { timeIntervalSincePreviousTimeInterval } : {})
+    }), 'requestState'], COMPLETED);
+    return out;
+  },
+
+  startAction: ({
+    pipeline,
+    mapping = defaultMapping,
+    timeIntervalSinceToday,
+    timeIntervalUnits,
+    timeIntervalSincePreviousTimeInterval
+  }) => {
+    const out = ({
+      pipeline,
+      mapping,
+      timeIntervalSinceToday,
+      timeIntervalUnits,
+      timeIntervalSincePreviousTimeInterval
+    });
+    return out;
+  },
+
+  successAction: (args) => {
+    if (!args) {
+      return {};
+    }
+    const { pipeline, timeIntervalSinceToday, timeIntervalUnits, timeIntervalSincePreviousTimeInterval, result, sinceAt } = args;
+    const out = ({ pipeline, timeIntervalSinceToday, timeIntervalUnits, timeIntervalSincePreviousTimeInterval, result, sinceAt });
+    return out;
+  },
+
+  failureAction: args => args,
+
+  completeAction: args => args,
+
+  checkShouldFire: ({
```

Comment:
Is this new? What calls this?

Reply:
It'll get called in the asyncDuck.
Since we're using ws instead, the behaviour of the async duck needs to be a bit different. It was probably continuously looping, due to initial ws rest request completing, but not having the results yet (as they're coming through via web sockets later).